### ISSUE TITLE
Improve documentation for read_env()

### ISSUE THREAD

**Issue by edmorley** [joke2k/django-environ#88]
The readme briefly mentions `read_env()`, however it wasn't until reading the source that the following became apparent:
* that the values read in from the `.env` file are overridden by any that already existed in the environment (I think this is handy, but isn't immediately obvious)
* that `read_env()` also takes an overrides list
* that `read_env()` updates `os.environ` directly, rather than just that particular `Env` instance (someone not realising this was intentional is what caused #66 to be filed)

It would be great to mention the above in the readme. If I get a chance to open a PR I will, but it may not be soon, so if anyone wants to do so in the meantime, go ahead :-)

**Comment by bittner** [comment]
Also, the documentation mentions a "[filthy magic stack backtracking](https://github.com/joke2k/django-environ/blob/develop/environ/environ.py#L595)" in order to "find manage.py and then find the dotenv".

It should be specified more in detail what this means. When you have a two-level hierarchy then the `.env` file is not found, at the moment. It's yet to be discussed whether this is by design or a bug.
``` ini
mysite
+-- mysite
|   +-- settings.py
+-- .env               # not found by Env.read_env() in mysite.settings
```

**Comment by dk-WZFinTech** [comment]
IMO less problematic will be to inform user that they can point path for `.env` file, like I did here:

```python
BASE_DIR = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))

env = environ.Env(
    DEBUG=(bool, False),
)

environ.Env.read_env('{}/.env'.format(BASE_DIR))
```

**Comment by bittner** [comment]
I'd suggest to use `os.path.join(base, ".env")` instead of a slash in a string. The former is OS independent.

**Comment by dk-WZFinTech** [comment]
> I'd suggest to use `os.path.join(base, ".env")` instead of a slash in a string. The former is OS independent.

True my bad :smile:

**Comment by remarkov** [comment]
> Also, the documentation mentions a "[filthy magic stack backtracking](https://github.com/joke2k/django-environ/blob/develop/environ/environ.py#L595)" in order to "find manage.py and then find the dotenv".
> 
> It should be specified more in detail what this means. When you have a two-level hierarchy then the `.env` file is not found, at the moment. It's yet to be discussed whether this is by design or a bug.
> 
> ```ini
> mysite
> +-- mysite
> |   +-- settings.py
> +-- .env               # not found by Env.read_env() in mysite.settings
> ```

That really gave me some wtf moments. After reading that wording regarding "filthy magic" I was literally expect it to magically find that my .env is located at the same directory as manage.py which was one level up from settings.py.
Thanks all for this discussion and hints.

**Comment by jbbqqf** [comment]
Hi! Triaging older issues — I think this one has shipped and can be closed.

Evidence:
- `docs/tips.rst` lines 472-478 ([overwrite section](https://github.com/joke2k/django-environ/blob/develop/docs/tips.rst#L472-L478)) explicitly documents that `read_env()` does *not* overwrite existing environment variables by default, plus the `overwrite=True` opt-in.
- `docs/faq.rst` lines 41-44 ([FAQ entry](https://github.com/joke2k/django-environ/blob/develop/docs/faq.rst#L41-L44)) explains the precedence rule in plain English.
- The `read_env()` docstring at [`environ/environ.py`:1058-1080](https://github.com/joke2k/django-environ/blob/develop/environ/environ.py#L1058-L1080) covers `overwrite`, `parse_comments`, `encoding`, and the `**overrides` kwarg.

If I'm wrong about coverage and something specific is still missing, point me at the gap and I'll dig further. Otherwise would you mind closing this out?

---
*Disclosure: I drafted this comment with help from Claude Code while triaging stale issues; the references above were verified manually.*
